File: py/branchdailyexports_sdk/features.py
# BranchDailyExports SDK feature factory

from branchdailyexports_sdk.feature.base_feature import BranchDailyExportsBaseFeature
from branchdailyexports_sdk.feature.debug_feature import BranchDailyExportsDebugFeature
from branchdailyexports_sdk.feature.idempotency_feature import BranchDailyExportsIdempotencyFeature
from branchdailyexports_sdk.feature.metrics_feature import BranchDailyExportsMetricsFeature
from branchdailyexports_sdk.feature.paging_feature import BranchDailyExportsPagingFeature
from branchdailyexports_sdk.feature.ratelimit_feature import BranchDailyExportsRatelimitFeature
from branchdailyexports_sdk.feature.retry_feature import BranchDailyExportsRetryFeature
from branchdailyexports_sdk.feature.test_feature import BranchDailyExportsTestFeature
from branchdailyexports_sdk.feature.timeout_feature import BranchDailyExportsTimeoutFeature


_FEATURES = {
    "base": lambda: BranchDailyExportsBaseFeature(),
    "debug": lambda: BranchDailyExportsDebugFeature(),
    "idempotency": lambda: BranchDailyExportsIdempotencyFeature(),
    "metrics": lambda: BranchDailyExportsMetricsFeature(),
    "paging": lambda: BranchDailyExportsPagingFeature(),
    "ratelimit": lambda: BranchDailyExportsRatelimitFeature(),
    "retry": lambda: BranchDailyExportsRetryFeature(),
    "test": lambda: BranchDailyExportsTestFeature(),
    "timeout": lambda: BranchDailyExportsTimeoutFeature(),
}


def _make_feature(name):
    factory = _FEATURES.get(name)
    if factory is not None:
        return factory()
    return _FEATURES["base"]()


# True when this SDK was generated with the named feature class - the
# constructor's tolerance for extend-carried features reads this (an
# active name with no generated class must not become a BaseFeature
# stray when an extend instance carries it).
def _has_feature(name):
    return name in _FEATURES

File: php/features.php
<?php
declare(strict_types=1);

// BranchDailyExports SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/DebugFeature.php';
require_once __DIR__ . '/feature/IdempotencyFeature.php';
require_once __DIR__ . '/feature/MetricsFeature.php';
require_once __DIR__ . '/feature/PagingFeature.php';
require_once __DIR__ . '/feature/RatelimitFeature.php';
require_once __DIR__ . '/feature/RetryFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';
require_once __DIR__ . '/feature/TimeoutFeature.php';


class BranchDailyExportsFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new BranchDailyExportsBaseFeature();
            case "debug":
                return new BranchDailyExportsDebugFeature();
            case "idempotency":
                return new BranchDailyExportsIdempotencyFeature();
            case "metrics":
                return new BranchDailyExportsMetricsFeature();
            case "paging":
                return new BranchDailyExportsPagingFeature();
            case "ratelimit":
                return new BranchDailyExportsRatelimitFeature();
            case "retry":
                return new BranchDailyExportsRetryFeature();
            case "test":
                return new BranchDailyExportsTestFeature();
            case "timeout":
                return new BranchDailyExportsTimeoutFeature();
            default:
                return new BranchDailyExportsBaseFeature();
        }
    }

    /**
     * Does a generated feature class back this name? False for a name only
     * an options extend instance can supply (the station adopt path) - the
     * constructor uses this to skip make_feature for such names instead of
     * adding a stray BaseFeature.
     */
    public static function has_feature(string $name): bool
    {
        switch ($name) {
            case "base":
            case "debug":
            case "idempotency":
            case "metrics":
            case "paging":
            case "ratelimit":
            case "retry":
            case "test":
            case "timeout":
                return true;
            default:
                return false;
        }
    }
}
